### PR TITLE
Remove unneeded args from objective getters

### PR DESCRIPTION
**Pull request by f0uriest**
- When using stuff like ``get_fixed_boundary_constraints`` we have to specify both whether to fix profiles, and also which profiles the equilibrium has.
- Now that ``eq`` is a required argument to the get_* functions, this now redundant, since we can just look and see what profiles the equilibrium has.
- This removes the unneeded args from all the get_* methods, so that the only things needed are the equilibrium and a bool for profiles, the specific profiles will be determined from the equilibrium.

### CODE REVIEW

File: tests/test_compute_funs.py [hunk 1]
```diff
@@ -1292,7 +1292,8 @@ def test_compute_everything():
                     np.testing.assert_allclose(
                         actual=this_branch_data[p][name],
                         desired=master_data[p][name],
-                        atol=1e-12,
+                        atol=1e-10,
```
Comment (f0uriest) [inline tests/test_compute_funs.py:1295]: This is needed due to minor differences caused by CPU types. Previously pretty much all the gh runners used intel xeon cpus, but recently they've upgraded some to the newer AMD Ryzen ones. This seems to lead to tiny differences in some computed quantities. For most things the difference is on the order of machine precision so no problem, but for ``current_r`` at the axis the difference is ~1e-11. The difference in ``I_r`` is ~1e-17, but when we scale by mu0 we get larger differences for ``current_r``

File: tests/test_examples.py [hunk 2]
```diff
@@ -548,12 +548,12 @@ def test_NAE_QSC_solve():
 
     # this has all the constraints we need,
     #  iota=False specifies we want to fix current instead of iota
```
Comment (dpanici) [inline tests/test_examples.py:550]: comment does not correspond to code now

File: tests/test_examples.py [hunk 3]
```diff
@@ -636,7 +636,7 @@ def test_NAE_QIC_solve():
...
```
Comment (dpanici) [inline tests/test_examples.py:638]: comment does not correspond to code now

**Review by dpanici — CHANGES_REQUESTED** [review]
2 minor comments
 - add to docstring of getter that it will use whatever profiles are present in the eq
 - a couple of comments in the code refer to iota=False when it is no longer present

**Review by ddudt — APPROVED** [review]
Looks good, besides the comments @dpanici already made.

File: desc/objectives/getters.py [hunk 4]
```diff
@@ -102,58 +105,24 @@ def get_fixed_axis_constraints(
...
-                ),
-                FixAtomicNumber(eq=eq, normalize=normalize, normalize_target=normalize),
-            )
-        else:
-            constraints += (
-                FixPressure(eq=eq, normalize=normalize, normalize_target=normalize),
-            )
-            if anisotropy:
+        for name, con in _PROFILE_CONSTRAINTS.items():
+            if getattr(eq, name) is not None:
```
Comment (rahulgaur104) [inline desc/objectives/getters.py:109]: So if the attribute is not in the dictionary, would it be free?

**Review by rahulgaur104 — APPROVED** [review]

File: desc/objectives/getters.py [hunk 5]
```diff
@@ -70,23 +81,15 @@ def get_equilibrium_objective(eq, mode="force", normalize=True):
...
+        Equilibrium to constrain.
     profiles : bool
-        Whether to also return constraints to fix input profiles.
-    iota : bool
-        Whether to add FixIota or FixCurrent as a constraint.
-    kinetic : bool
-        Whether to add constraints to fix kinetic profiles or pressure
-    anisotropy : bool
-        Whether to add constraint to fix anisotropic pressure.
+        If True, also include constraints to fix all profiles assigned to equilibrium.
```
[the remark this answers is not in the document]
Comment (dpanici) [reply desc/objectives/getters.py:92]: I think you should just create the constraints yourself then for something like that, this getter is mainly for the typical use case of fixing profiles, Psi, and boundary

File: desc/objectives/getters.py (hunk 4 above)
Comment (f0uriest) [reply desc/objectives/getters.py:109]: the dictionary covers all profiles types, so it should fix anything that the equilibrium has

Comment (dpanici) [reply desc/objectives/getters.py:109]: Not free, if an equilibrium does not have that profile assigned, then it does not have that attribute at all

i.e. the `FixIota` constraint will not be given to an equilibrium which does not have an `eq.iota` attribute, but if it lacks that attribute that means that instead it has a `eq.current` attribute instead and that one will be fixed

same for kinetic profiles(n and T) and pressure, if the former are present then the latter must not be present, and vice-versa

**Pull request merged**